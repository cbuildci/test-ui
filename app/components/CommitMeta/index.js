/*
 * CommitMeta
 */

import React from 'react';
import PropTypes from 'prop-types';

function CommitMeta({
    githubHost,
    message,
    author,
}) {
    return (
        <div>
            {author.login && (
                <span className="mr-1" style={{ lineHeight: '22px' }}>
                    <img
                        src={`https://github.com/${author.login}.png`}
                        width="20"
                        height="20"
                        style={{ marginTop: '-2px', borderRadius: '4px' }}
                    />
                </span>
            )}
            <span className="mr-2 flex-shrink-0">
                {author.login ? (
                    <a href={`https://${githubHost}/${author.login}`}>
                        {author.login}
                    </a>
                ) : (
                    <span>
                        {author.name}
                    </span>
                )}
            </span>

            <span className="mr-2 text-nowrap text-muted" style={{ overflow: 'hidden', textOverflow: 'ellipsis' }}>
                – {message.substr(0, 100)}{message.length > 100 ? '…' : ''}
            </span>
        </div>
    );
}

CommitMeta.propTypes = {
    githubHost: PropTypes.string.isRequired,
    message: PropTypes.string.isRequired,
    author: PropTypes.object.isRequired,
};

export default CommitMeta;
