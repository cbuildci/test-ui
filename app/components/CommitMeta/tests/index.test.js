import React from 'react';
import { snapshots } from '../../../../internals/testing/snapshot-util';
import CommitMeta from '../index';

describe('<CommitMeta />', () => {
    it('should render expected JSX', () => {
        snapshots(
            <CommitMeta
                githubHost="foohub.com"
                message="abc123"
                author={{
                    login: 'foobar',
                    name: 'foo',
                    email: 'bar',
                }}
            />,
        );
    });

    it('should render expected JSX if author is missing login', () => {
        snapshots(
            <CommitMeta
                githubHost="foohub.com"
                message="abc123"
                author={{
                    name: 'foo',
                    email: 'bar',
                }}
            />,
        );
    });

    it('should render expected JSX if message is exactly 100 characters', () => {
        snapshots(
            <CommitMeta
                githubHost="foohub.com"
                message={'1234567890'.repeat(10)}
                author={{
                    login: 'foobar',
                    name: 'foo',
                    email: 'bar',
                }}
            />,
        );
    });

    it('should render expected JSX if message is over 100 characters', () => {
        snapshots(
            <CommitMeta
                githubHost="foohub.com"
                message={'1234567890'.repeat(10) + 'A'}
                author={{
                    login: 'foobar',
                    name: 'foo',
                    email: 'bar',
                }}
            />,
        );
    });
});
