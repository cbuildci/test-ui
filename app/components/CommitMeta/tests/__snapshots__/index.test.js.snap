// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommitMeta /> should render expected JSX 1`] = `
<div>
  <span
    className="mr-1"
    style={
      Object {
        "lineHeight": "22px",
      }
    }
  >
    <img
      height="20"
      src="https://github.com/foobar.png"
      style={
        Object {
          "borderRadius": "4px",
          "marginTop": "-2px",
        }
      }
      width="20"
    />
  </span>
  <span
    className="mr-2 flex-shrink-0"
  >
    <a
      href="https://foohub.com/foobar"
    >
      foobar
    </a>
  </span>
  <span
    className="mr-2 text-nowrap text-muted"
    style={
      Object {
        "overflow": "hidden",
        "textOverflow": "ellipsis",
      }
    }
  >
    – 
    abc123
  </span>
</div>
`;

exports[`<CommitMeta /> should render expected JSX 2`] = `
<div>
  <span
    class="mr-1"
    style="line-height:22px"
  >
    <img
      height="20"
      src="https://github.com/foobar.png"
      style="margin-top:-2px;border-radius:4px"
      width="20"
    />
  </span>
  <span
    class="mr-2 flex-shrink-0"
  >
    <a
      href="https://foohub.com/foobar"
    >
      foobar
    </a>
  </span>
  <span
    class="mr-2 text-nowrap text-muted"
    style="overflow:hidden;text-overflow:ellipsis"
  >
    – abc123
  </span>
</div>
`;

exports[`<CommitMeta /> should render expected JSX if author is missing login 1`] = `
<div>
  <span
    className="mr-2 flex-shrink-0"
  >
    <span>
      foo
    </span>
  </span>
  <span
    className="mr-2 text-nowrap text-muted"
    style={
      Object {
        "overflow": "hidden",
        "textOverflow": "ellipsis",
      }
    }
  >
    – 
    abc123
  </span>
</div>
`;

exports[`<CommitMeta /> should render expected JSX if author is missing login 2`] = `
<div>
  <span
    class="mr-2 flex-shrink-0"
  >
    <span>
      foo
    </span>
  </span>
  <span
    class="mr-2 text-nowrap text-muted"
    style="overflow:hidden;text-overflow:ellipsis"
  >
    – abc123
  </span>
</div>
`;

exports[`<CommitMeta /> should render expected JSX if message is exactly 100 characters 1`] = `
<div>
  <span
    className="mr-1"
    style={
      Object {
        "lineHeight": "22px",
      }
    }
  >
    <img
      height="20"
      src="https://github.com/foobar.png"
      style={
        Object {
          "borderRadius": "4px",
          "marginTop": "-2px",
        }
      }
      width="20"
    />
  </span>
  <span
    className="mr-2 flex-shrink-0"
  >
    <a
      href="https://foohub.com/foobar"
    >
      foobar
    </a>
  </span>
  <span
    className="mr-2 text-nowrap text-muted"
    style={
      Object {
        "overflow": "hidden",
        "textOverflow": "ellipsis",
      }
    }
  >
    – 
    1234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890
  </span>
</div>
`;

exports[`<CommitMeta /> should render expected JSX if message is exactly 100 characters 2`] = `
<div>
  <span
    class="mr-1"
    style="line-height:22px"
  >
    <img
      height="20"
      src="https://github.com/foobar.png"
      style="margin-top:-2px;border-radius:4px"
      width="20"
    />
  </span>
  <span
    class="mr-2 flex-shrink-0"
  >
    <a
      href="https://foohub.com/foobar"
    >
      foobar
    </a>
  </span>
  <span
    class="mr-2 text-nowrap text-muted"
    style="overflow:hidden;text-overflow:ellipsis"
  >
    – 1234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890
  </span>
</div>
`;

exports[`<CommitMeta /> should render expected JSX if message is over 100 characters 1`] = `
<div>
  <span
    className="mr-1"
    style={
      Object {
        "lineHeight": "22px",
      }
    }
  >
    <img
      height="20"
      src="https://github.com/foobar.png"
      style={
        Object {
          "borderRadius": "4px",
          "marginTop": "-2px",
        }
      }
      width="20"
    />
  </span>
  <span
    className="mr-2 flex-shrink-0"
  >
    <a
      href="https://foohub.com/foobar"
    >
      foobar
    </a>
  </span>
  <span
    className="mr-2 text-nowrap text-muted"
    style={
      Object {
        "overflow": "hidden",
        "textOverflow": "ellipsis",
      }
    }
  >
    – 
    1234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890
    …
  </span>
</div>
`;

exports[`<CommitMeta /> should render expected JSX if message is over 100 characters 2`] = `
<div>
  <span
    class="mr-1"
    style="line-height:22px"
  >
    <img
      height="20"
      src="https://github.com/foobar.png"
      style="margin-top:-2px;border-radius:4px"
      width="20"
    />
  </span>
  <span
    class="mr-2 flex-shrink-0"
  >
    <a
      href="https://foohub.com/foobar"
    >
      foobar
    </a>
  </span>
  <span
    class="mr-2 text-nowrap text-muted"
    style="overflow:hidden;text-overflow:ellipsis"
  >
    – 1234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890…
  </span>
</div>
`;
